cmd/csi-snapshotter/main.go

```diff
@@ -51,7 +51,7 @@ const (
 	threads = 10
 
 	// Default timeout of short CSI calls like GetPluginInfo
-	csiTimeout = time.Second
+	defaultCSITimeout = time.Second
```

Comment:
We should probably optimize the default for the primary operations like createSnapshot instead of the metadata operations.

Reply:
Makes sense, any suggestions for what this should be? 5s?

Reply:
@xing-yang do you have any ideas on what would be a good number?

external-provisioner uses 10 seconds, external-attacher uses 15 seconds.

Reply:
This depends on the storage backend.  We can set the default to 10 seconds, same as external-provisioner.

Reply:
set to 10s

Reply:
@andrewsykim 
I saw couple of cases (GKE regional, Netapp etc) took more than 10s for cutting snapshot (between 30-45 s). 
If we set up default timeout to a higher value, a default experience for users will be better since they have lower chance to see false errors. Of course, it will have some drawbacks such as waiting unnecessarily for the timeouts. However, I still think that positive parts are more important than the drawbacks I mentioned.

Reply:
I do think 30s is more realistic.  I suggested 10s because the default for external-provisioner is 10s, however, creating a volume usually takes much longer time than cutting a snapshot.

@hakanmemisoglu Have you tested this fix?  Does it address your problem now?

Reply:
I had a basic fix where I was setting it to 1min (which is excess). I will update the comment when I try this one.

Reply:
The downside of waiting for a snapshot to be cut for unnecessarily long time is that it may not be desirable for an application if it is frozen before the snapshotting.

Reply:
How do the snapshot timeouts work? Do the users specify their desired timeout value?  Snapshots timeout handling is trickier than the other sidecars because the other sidecars will retry forever, whereas snapshots are only tried once.

Reply:
@xing-yang @andrewsykim 
It works perfectly.

Tested with:
- Helm repo: stable/mysql
- PVC size: 8GiB (with almost no data)
- GKE 1.13.4-gke.10 (Public preview)
- Non-regional snapshot

Reply:
Thanks @hakanmemisoglu !

Reply:
So what's the consensus for default time out? 30s?

Reply:
@msau42 User will be specifying the desired timeout value using this timeout parameter.  In the ExecutionHook we are still designing, there will be other timeout value to control how long we should wait between the start of a freeze and the start of a thaw.

You made a good point that other sidecars will retry while snapshot does not.  Given that, it is better to set default to a longer time.  I looked at the old code, we set this to 1 minute before the recent change.  I’d prefer setting default to the original value of 1 minute.  Otherwise driver users will be surprised when snapshotting starts to fail with a much shorter default timeout value.
https://github.com/kubernetes-csi/external-snapshotter/blob/release-1.0/cmd/csi-snapshotter/main.go#L54

Reply:
sgtm, updating to 1m